harmony_regridding_service/grid.py

```diff
@@ -59,78 +62,57 @@ def compute_target_area(
-    exist, they are created.
+    exist, they are created from the source grid.
 
     """
     if has_scale_extents(message) and (
         has_scale_sizes(message) or has_dimensions(message)
     ):
-        return get_grid_parameters_from_message(message)
-    return create_grid_parameters_from_source(filepath, var_info)
+        return get_area_definition_from_message(message)
 
+    if same_source_and_target_crs(var_info):
+        raise SameSourceTargetCRS('The requested CRS is the same as the source CRS.')
 
-def create_grid_parameters_from_source(
-    filepath: str,
+    return create_area_definition_from_source(filepath, var_info)
+
+
+def same_source_and_target_crs(
     var_info: VarInfoFromNetCDF4,
-) -> tuple[tuple, int, int]:
-    """Create the target grid parameters using the source grid information.
-
-    The area extent is taken from the source granule's upper right and lower
-    left corners. While area extents are typically defined using the outermost
-    points of the cell, the location of the geographic point within the cell
-    can vary per collection and cannot be assumed to be in the center of the
-    cell, so calulating the cell extents cannot easily be generalized.
-
-    This grid parameter calculation is an estimate given the limited information
-    received from the user. The differences between just using the input
-    grid's min/max values versus calculating the exact outermost corner points is
-    considered to be minimal.
-    """
-    dimension_pairs = get_resampled_dimension_pairs(var_info)
-    variables = get_variables_for_dimension_pair(dimension_pairs[0], var_info)
-    source_swath = compute_source_swath(
-        dimension_pairs[0], filepath, var_info, variables
-    )
+) -> bool:
+    """Check if the requested CRS is the same as the input CRS.
 
-    column_min = source_swath.lons[-1, 0]
-    row_min = source_swath.lats[-1, 0]
-    column_max = source_swath.lons[0, -1]
-    row_max = source_swath.lats[0, -1]
+    For now, it is assumed that only one geographic CRS is available, but this
+    may not be the case in the future.
+    """
+    grid_dimensions = get_resampled_dimension_pairs(var_info)[0]
+    return dims_are_lon_lat(grid_dimensions, var_info)
 
-    area_extent = (column_min, row_min, column_max, row_max)
 
-    width = source_swath.shape[1]
-    height = source_swath.shape[0]
+def create_area_definition_from_source(
+    filepath: str,
+    var_info: VarInfoFromNetCDF4,
+) -> AreaDefinition:
+    """Create the target area definition using the source grid information.
 
-    return area_extent, height, width
+    TODO: Create area definition for every dimension pair, if the collection
+    had more than one grid.
+    """
+    dimension_pairs = get_resampled_dimension_pairs(var_info)
+    return create_area_definition(filepath, dimension_pairs[0], var_info)
 
 
 def get_variables_for_dimension_pair(dimpair, var_info):
```

Comment:
I'm still not convinced this is a great helper function. But for now, at least rename `dimpair` -> `dim_pair`
```suggestion
def get_variables_for_dimension_pair(dim_pair, var_info):
```

Reply:
Oh right, I remember we did mention this on Friday. I see no issue with taking the function out altogether for now, since it's only used in one place and is essentially just a line of code?